### ISSUE TITLE
definemod that's not player specific

### ISSUE THREAD

**Issue by XeroOl** [XeroOl/notitg-mirin#51]
I wanr to make a way to create non-player-specific definemods, and change the way definemods work to be more intuitive (potentially a different aux system).

Making this change would require rewriting the whole node system, whicj I want to do anyway.

**Comment by XeroOl** [comment]
I remember discussing the details of how this would work, but I don't remember them now.

**Comment by XeroOl** [comment]
```lua
definemod{'foo', 'bar', 'pn', function(foo, bar, pn)

end}```
`pn` will become a magical keyword for definemod/node that changes the definition from a normal one into a player specific one.
If a node:
1) depends on a mod that needs a pn (ie, 'invert'), or
2) writes to a player specific mod (ie 'invert')
then your definemod would automatically be promoted to be player specific, without needing to specifically request the pn.

Under these conditions, definemods would behave like this:
```lua
-- Not player specific
definemod {'bg_pos', function(x) bg:x(x) end}
definemod {'objr', objangle', function(r, a) return r*math.cos(a), r*math.sin(a) end, 'objx', 'objy'}
definemod {'objx', 'objy', function(x, y) obj:xy(x, y) end}


-- Are player specific
definemod {'games', -75, 'invert', 25, 'flip'}  -- because it writes to player specific mods
aux {'claims'}; node {'claims', 'invert', function(c, i) end} -- because it reads from player specific mods
definemod {'flames', 100, 'james'} -- because it interacts with 'james'' definemod, which needs pn 
definemod {'james', 'pn', function(j, pn) P[pn]:xy(j, j) end} -- because it explicitly requests pn
```

**Comment by XeroOl** [comment]
I don't know if those semantics are consistent / possible to implement. I will have to think strongly through it. I'm concerned that I'm attaching the player specific-ness to the node, when it really needs to be a property of each mod

**Comment by ChocoboGamer** [comment]
Already half implemented node refactor has been finished, the next step is solving this commit.

**Comment by ChocoboGamer** [comment]
Closed as duplicate of #85
